jupyter_core/tests/test_paths.py

```diff
@@ -99,159 +99,211 @@ def test_envset():
+def test_config_dir():
+    config = jupyter_config_dir()
     assert config == home_jupyter
 
-    with darwin, config_env:
-        config = jupyter_config_dir()
-    assert config == jupyter_config_env
+
+@macos
+@use_platformdirs
+def test_config_dir_darwin():
+    config = jupyter_config_dir()
+    assert config == realpath("~/Library/Preferences/Jupyter")
 
 
-@pytest.mark.skipif(sys.platform != "win32", reason="only run on windows")
+@windows
+@use_platformdirs
 def test_config_dir_windows():
-    with no_config_env:
-        config = jupyter_config_dir()
-    assert config == home_jupyter
+    config = jupyter_config_dir()
+    assert config == realpath(pjoin(os.environ.get("LOCALAPPDATA", ""), "Jupyter"))
+
+
+@linux
+@use_platformdirs
+def test_config_dir_linux():
+    config = jupyter_config_dir()
+    assert config == realpath("~/.config/jupyter")
 
+
+def test_config_env_legacy():
     with config_env:
         config = jupyter_config_dir()
-    assert config == jupyter_config_env
+        assert config == jupyter_config_env
 
 
-@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
-def test_config_dir_linux():
-    with linux, no_config_env:
+@use_platformdirs
+def test_config_env():
+    with config_env:
         config = jupyter_config_dir()
-    assert config == home_jupyter
+        assert config == jupyter_config_env
 
-    with linux, config_env:
-        config = jupyter_config_dir()
-    assert config == jupyter_config_env
+
+def test_data_dir_env_legacy():
+    data_env = "runtime-dir"
+    with patch.dict("os.environ", {"JUPYTER_DATA_DIR": data_env}):
+        data = jupyter_data_dir()
+        assert data == data_env
 
 
+@use_platformdirs
 def test_data_dir_env():
     data_env = "runtime-dir"
     with patch.dict("os.environ", {"JUPYTER_DATA_DIR": data_env}):
         data = jupyter_data_dir()
-    assert data == data_env
+        assert data == data_env
 
 
-@pytest.mark.skipif(sys.platform == "win32", reason="does not run on windows")
-def test_data_dir_darwin():
-    with darwin:
-        data = jupyter_data_dir()
+@macos
+def test_data_dir_darwin_legacy():
+    data = jupyter_data_dir()
     assert data == realpath("~/Library/Jupyter")
 
-    with darwin, xdg:
-        # darwin should ignore xdg
-        data = jupyter_data_dir()
-    assert data == realpath("~/Library/Jupyter")
 
+@macos
+@use_platformdirs
+def test_data_dir_darwin():
+    data = jupyter_data_dir()
+    assert data == realpath("~/Library/Application Support/Jupyter")
 
-@pytest.mark.skipif(sys.platform != "win32", reason="only run on windows")
-def test_data_dir_windows():
+
+@windows
+def test_data_dir_windows_legacy():
     data = jupyter_data_dir()
     assert data == realpath(pjoin(os.environ.get("APPDATA", ""), "jupyter"))
 
+
+@windows
+@use_platformdirs
+def test_data_dir_windows():
+    data = jupyter_data_dir()
+    assert data == realpath(pjoin(os.environ.get("LOCALAPPDATA", ""), "Jupyter"))
```

Comment:
Is there a clear reason why we should switch windows from APPDATA (which in our tests is AppData/Roaming) to the local appdata (AppData/Local)? We can switch back to remote with the platformdirs remote=True argument.

I don't know Windows well enough to know if there is a strong clear reason we should move to local appdata over just generic appdata.

Reply:
https://learn.microsoft.com/en-us/windows/apps/design/app-settings/store-and-retrieve-app-data gives some background, including the fact that roaming data is no longer supported as of Windows 11, apparently. So there's a strong reason to move to local appdata. Also roaming data has a pretty severe quota apparently.